Description: Let the build system honor flags obtained through dpkg-buildflags.
 Try to append flags where possible. Also remove the -Ox flags set by upstream.
Origin: vendor
Forwarded: http://sourceforge.net/tracker/?func=detail&aid=3470774&group_id=132964&atid=725711
Author: Cristian Greco <cristian@debian.org>

--- a/build/config/Linux
+++ b/build/config/Linux
@@ -20,7 +20,7 @@
 LINK    = $(CXX)
 LIB     = ${CROSS_COMPILE}ar -cr
 RANLIB  = ${CROSS_COMPILE}ranlib
-SHLIB   = $(CXX) -shared -Wl,-soname,$(notdir $@) -o $@
+SHLIB   = $(CXX) -shared -Wl,-soname,$(notdir $@) $(LDFLAGS) -o $@
 SHLIBLN = $(POCO_BASE)/build/script/shlibln
 STRIP   = ${CROSS_COMPILE}strip
 DEP     = $(POCO_BASE)/build/script/makedepend.gcc 
@@ -38,13 +38,13 @@
 #
 # Compiler and Linker Flags
 #
-CFLAGS          = 
+CFLAGS         +=
 CFLAGS32        =
 CFLAGS64        =
-CXXFLAGS        = -Wall -Wno-sign-compare
+CXXFLAGS       += -Wall -Wno-sign-compare
 CXXFLAGS32      =
 CXXFLAGS64      =
-LINKFLAGS       =
+LINKFLAGS       = $(LDFLAGS)
 LINKFLAGS32     =
 LINKFLAGS64     =
 STATICOPT_CC    =
@@ -56,14 +56,14 @@
 DEBUGOPT_CC     = -g -D_DEBUG
 DEBUGOPT_CXX    = -g -D_DEBUG
 DEBUGOPT_LINK   = -g
-RELEASEOPT_CC   = -O2 -DNDEBUG
-RELEASEOPT_CXX  = -O2 -DNDEBUG
-RELEASEOPT_LINK = -O2
+RELEASEOPT_CC   = -DNDEBUG
+RELEASEOPT_CXX  = -DNDEBUG
+RELEASEOPT_LINK =
 
 #
 # System Specific Flags
 #
-SYSFLAGS = -D_XOPEN_SOURCE=500 -D_REENTRANT -D_THREAD_SAFE -D_FILE_OFFSET_BITS=64 -D_LARGEFILE64_SOURCE -DPOCO_HAVE_FD_EPOLL
+SYSFLAGS = -D_XOPEN_SOURCE=500 -D_REENTRANT -D_THREAD_SAFE -D_FILE_OFFSET_BITS=64 -D_LARGEFILE64_SOURCE -DPOCO_HAVE_FD_EPOLL $(CPPFLAGS)
 
 #
 # System Specific Libraries
--- a/build/config/FreeBSD
+++ b/build/config/FreeBSD
@@ -20,7 +20,7 @@
 LINK    = $(CXX)
 LIB     = ar -cr
 RANLIB  = ranlib
-SHLIB   = $(CXX) -shared -Wl,-soname,$(notdir $@) -o $@
+SHLIB   = $(CXX) -shared -Wl,-soname,$(notdir $@) $(LDFLAGS) -o $@
 SHLIBLN = $(POCO_BASE)/build/script/shlibln
 DEP     = $(POCO_BASE)/build/script/makedepend.gcc 
 STRIP   = strip
@@ -38,13 +38,13 @@
 #
 # Compiler and Linker Flags
 #
-CFLAGS          = 
+CFLAGS         +=
 CFLAGS32        =
 CFLAGS64        =
-CXXFLAGS        =
+CXXFLAGS       +=
 CXXFLAGS32      =
 CXXFLAGS64      =
-LINKFLAGS       =
+LINKFLAGS       = $(LDFLAGS)
 LINKFLAGS32     =
 LINKFLAGS64     =
 STATICOPT_CC    =
@@ -56,14 +56,14 @@
 DEBUGOPT_CC     = -g -D_DEBUG
 DEBUGOPT_CXX    = -g -D_DEBUG
 DEBUGOPT_LINK   = -g
-RELEASEOPT_CC   = -O2 -DNDEBUG
-RELEASEOPT_CXX  = -O2 -DNDEBUG
-RELEASEOPT_LINK = -O2
+RELEASEOPT_CC   = -DNDEBUG
+RELEASEOPT_CXX  = -DNDEBUG
+RELEASEOPT_LINK =
 
 #
 # System Specific Flags
 #
-SYSFLAGS = -D_REENTRANT -D_THREAD_SAFE 
+SYSFLAGS = -D_REENTRANT -D_THREAD_SAFE $(CPPFLAGS)
 
 #
 # System Specific Libraries
